File: scripts/test_solutions/202_happy_number.js
var isHappy = function(n) {
    const seen = new Set();
    while (n !== 1 && !seen.has(n)) {
        seen.add(n);
        let sum = 0;
        while (n > 0) { sum += (n % 10) ** 2; n = Math.floor(n / 10); }
        n = sum;
    }
    return n === 1;
};

File: scripts/test_solutions/202_happy_number.py
def isHappy(n):
    seen = set()
    while n != 1 and n not in seen:
        seen.add(n)
        total = 0
        while n > 0:
            total += (n % 10) ** 2
            n //= 10
        n = total
    return n == 1
